### PR TITLE
Add tests for empty object behavior keys

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

The engine fails if a struct-behavior key has no fields.
This PR adds two tests for testing this behavior: One for an empty array (fixed-sized arrays are not allowed to have 0 elements), and one for structs with no fields. The latter test is currently disabled because of the bug mentioned above.

## 🔍 What does this change?

- Test for empty object: 325f0d106423f7c5a11b7484224ed319ed7b4a15
- Test for empty array: 4491a9c9475ec0aff2d22c0dbe7c4f161171d0c5

## 🔗 Related links

- [Asana bug](https://app.asana.com/0/1199548034582004/1201826554813225/f) (_internal_)

### Does this require a change to the docs?

No, but the follow-up PR should probably do when changing the behavior or adding a decent error message.

## 🛡 What tests cover this?

These are the tests, one is currently ignored because of the bug.

### CODE REVIEW

File: packages/engine/tests/units/state/empty/array/src/behaviors/test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Verifies the type of the list
+ */
+const behavior = (state, context) => {
+  state.l1_is_list = typeof state.l1 === "object";
```
Comment (Alfred-Mountfield) [inline packages/engine/tests/units/state/empty/array/src/behaviors/test.js:5]: Does `Array.isArray` work?

File: packages/engine/tests/units/state/empty/object/src/behaviors/test.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Verifies the type of the object
+ */
+const behavior = (state, context) => {
+  state.o1_is_struct = typeof state.o1 === "object";
```
Comment (Alfred-Mountfield) [inline packages/engine/tests/units/state/empty/object/src/behaviors/test.js:5]: I think this test could use some improving, JS sucks and this isn't really checking it's a serialised arrow struct (i.e. a standard JS object)

https://stackoverflow.com/a/8511350/14687716

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]

File: packages/engine/tests/units/state/empty/array/src/behaviors/test.js (hunk 1 above)
Comment (TimDiekmann) [reply packages/engine/tests/units/state/empty/array/src/behaviors/test.js]: I changed the check to `Array.isArray(state.l1)`

File: packages/engine/tests/units/state/empty/object/src/behaviors/test.js (hunk 2 above)
Comment (Alfred-Mountfield) [reply packages/engine/tests/units/state/empty/object/src/behaviors/test.js:5]: Can you create a follow-up task in the backlog please

Comment (TimDiekmann) [reply packages/engine/tests/units/state/empty/object/src/behaviors/test.js:5]: [Link to Asana task](https://github.com/hashintel/hash/pull/299#discussion_r805808833) (_internal_)
